lib/src/source/hosted.dart

```diff
@@ -638,12 +648,18 @@ class _OfflineHostedSource extends BoundHostedSource {
     List<PackageId> versions;
     if (dirExists(dir)) {
       versions = listDir(dir)
+          // Only directories with a pubspec.yaml should be considered packages.
+          .where(_looksLikePackageDir)
           .map((entry) {
             var components = p.basename(entry).split('-');
             if (components.first != ref.name) return null;
-            return source.idFor(
-                ref.name, Version.parse(components.skip(1).join('-')),
-                url: _serverFor(ref.description));
+            try {
+              return source.idFor(
+                  ref.name, Version.parse(components.skip(1).join('-')),
+                  url: _serverFor(ref.description));
+            } on FormatException {
+              // Failing to parse a package-name is not bad.
```

Comment:
Could also be version that fails, but I agree... Failure to parse here is best ignored